src/utils.jl

```diff
@@ -0,0 +1,158 @@
+# JuliaText TextAnalysis.jl Utility Functions
+
+function jackknife_avg(scores)
+
+    #= The jackknife is a resampling technique especially useful for variance and bias estimation. 
+    Currently being used for averaging in ROUGE scores in evaluate.jl
+    :param scores: List of integers to average
+    :type scores: Array{Int64,1} =#
+    
+    if length(collect(Set(scores))) == 1
+    
+        #= In case the elements of the array are all equal=#
+        return scores[1]
+        
+    else
+    
+        #=store the maximum scores
+        from the m different sets of m-1 scores.
+        such that m is the len(score_list)=#
+        
+        average = []
+        
+        for i in scores
+            # dummy : list a particular combo of m-1 scores
+            dummy = [j for j in scores if i != j]
+            append!(average, max(dummy...))
+            end
+
+        return sum(average)/length(average)
+
+    end
+end
+
+function listify_ngrams(ngram_doc)
+    flattened = []
+    for item in ngram_doc
+        for i in 1:item[2]
+            push!(flattened, item[1])
+        end
+    end
+    return flattened
+end
+
+function weighted_lcs(X, Y, weighted = true, return_string = false, f = sqrt)
+    #=This function returns the longest common subsequence
+    of two strings using the dynamic programming algorithm.
+     param X : first string in tokenized form
+    type (X) : Array{SubString{String},1}
+     param Y : second string in tokenized form
+    type (Y) : Array{SubString{String},1}
+     param weighted : Weighted LCS is done if weighted is True (default)
+    type (weighted) : Boolean
+     param return_string : Function returns weighted LCS length when set to False (default).
+                           Function returns longest common substring when set to True.
+    type (return_string) : Boolean
+     param f: weighting function. The weighting function f must have the property
+              that f(x+y) > f(x) + f(y) for any positive integers x and y. 
+    type (f) : generic function which takes a float as an input and returns a float.
+    =#
+    
+    m, n = length(X), length(Y)
+    c_table = [zeros(n+1) for i in 1:m+1]
+    w_table = [zeros(n+1) for i in 1:m+1]
+    
+    for i in 1:(m+1)
+        
+        for j in 1:(n+1)
+            
+            if i == 1 || j == 1
+                continue
```

Comment:
A better way to handle this case should be starting the loop from index of 2 instead.

Reply:
Needed for another function later on. 
Will be able to use one function to perform three functions using different arguments 😛